### PR TITLE
Returndatacopy op code spec

### PR DESCRIPTION
**Pull request by smtmfft**

### CODE REVIEW

File: specs/opcode/3eRETURNDATACOPY.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   - `memory_offset` is at the 1st position of the stack
+   - `data_offset` is at the 2nd position of the stack
+   - `size` is at the 3rd position of the stack
+   - `last_callee_id` is in last callee context
+   - `return_data_offset` is in last callee context
+   - `return_data_size` is in last callee context
+
+## Exceptions
+
+1. stack overflow: stack is full, stack pointer = 0
```
Comment (Brechtpd) [inline specs/opcode/3eRETURNDATACOPY.md]: Should be stack underflow I think (values are only popped from the stack): 
`stack underflow: 1022 <= stack_pointer <= 1024`

File: specs/opcode/3eRETURNDATACOPY.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# RETURNDATACOPY opcode
+
+## Procedure
+
+The `RETURNDATACOPY` opcode pops `dest_offset`, `offset` and `size` from the stack.
+It then copies `size` bytes of return data in the current environment from an offset `offset` to the memory at the address `return_data_offset`. For out-of-bound scenarios where `size > return_data_size - offset`, EVM reverts the current context.
```
Comment (Brechtpd) [inline specs/opcode/3eRETURNDATACOPY.md]:
```suggestion
It then copies `size` bytes of return data in the current environment from `return_data_offset` at an `offset` to the memory at address `dest_offset`. For out-of-bound scenarios where `size > return_data_size - offset`, EVM reverts the current context.
```

File: specs/opcode/3eRETURNDATACOPY.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# RETURNDATACOPY opcode
+
+## Procedure
```
Comment (Brechtpd) [inline specs/opcode/3eRETURNDATACOPY.md:3]: Any special behavior to document when `is_root`, etc... ?

Comment (smtmfft) [reply specs/opcode/3eRETURNDATACOPY.md:3]: I think is_root has nothing to do with the returndatacopy, as it depends only on the previous call. If there is no previous call (no matter current call is root or not), returndatacopy gets 0 as its boundary (evm checks returndatasize which is 0 here), if param length is also 0, everything is still ok, if length > 0, then the context reverts as expected.
I added few words mentioned the case above in evm behavior part .

**Comment by smtmfft** [comment]
PSE Merged

**Pull request closed without merging**